lib/types.js

```diff
@@ -2,11 +2,44 @@ const isNumber = val => typeof val === 'number';
 
 const castToNumber = val => {
```

Comment:
you should move all your isFunctions to the top of the file. They're very similar functions so it makes sense to group them together and makes the code/file easier to read.